Scripts/Runtime/Data/ID/ID.cs

```diff
@@ -0,0 +1,17 @@
+using UnityEngine;
+
+namespace Anvil.Unity.Data
+{
+    /// <summary>
+    /// Resets the <see cref="CSharp.Data.ID"/> static class on load.
+    /// This allows for Unity's domain reloading to be disabled which speeds up Burst compiler development.
+    /// </summary>
+    internal static class ID
```

Comment:
Can we call this `StaticStateResetHelper` to be consistent with the one in Logging?